### PR TITLE
test: add e2e tests for css modules

### PR DESCRIPTION
**Pull request by negreirosleo**
This PR relates to #186. It adds e2e tests for CSS modules.
I used `expect .toContain('class-name')` because I was not sure if the generated name would always be the same.

### CODE REVIEW

File: e2e/fixtures/rsc-css-modules/package.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  "version": "0.1.0",
+  "type": "module",
+  "private": true,
+  "scripts": {
+    "dev": "waku dev",
+    "build": "waku build",
+    "start": "waku start"
+  },
+  "dependencies": {
+    "react": "19.0.0-canary-e3ebcd54b-20240405",
```
Comment (dai-shi) [inline e2e/fixtures/rsc-css-modules/package.json:12]: We are using the newer version. Please check `main` branch.

**Comment by dai-shi** [comment]
@himself65 Hi, would you be able to give a quick review please?

File: e2e/fixtures/rsc-css-modules/README.md [hunk 2]
```diff
@@ -0,0 +1,3 @@
+# RSC CSS Modules
```
Comment (himself65) [inline e2e/fixtures/rsc-css-modules/README.md:1]: update description

File: e2e/fixtures/rsc-css-modules/src/components/App.tsx [hunk 3]
```diff
@@ -0,0 +1,17 @@
+import styles from './app.module.css';
+import { ClientCounter } from './ClientCounter.js';
+
+const App = ({ name }: { name: string }) => {
+  // console.log(styles);
```
Comment (himself65) [inline e2e/fixtures/rsc-css-modules/src/components/App.tsx:5]: remove console

**Review by himself65 — CHANGES_REQUESTED** [review]

**Comment by himself65** [comment]
> @himself65 Hi, would you be able to give a quick review please?

btw, I was trying to add chromatic for visual test last year, forget to continue on this

File: e2e/fixtures/rsc-css-modules/README.md (hunk 2 above)
Comment (negreirosleo) [reply e2e/fixtures/rsc-css-modules/README.md:1]: is it good enough now?

**Comment by dai-shi** [comment]
Can you check CI errors? Are they understandable?

**Comment by negreirosleo** [comment]
> Can you check CI errors? Are they understandable?

I'm checking it, but I'm having a different error on my machine:
image

am I missing anything? I followed all the `CONTRIBUTING.MD` steps

File: e2e/fixtures/rsc-css-modules/package.json [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  },
+  "dependencies": {
+    "react": "19.0.0-beta-4508873393-20240430",
+    "react-dom": "19.0.0-beta-4508873393-20240430",
+    "react-server-dom-webpack": "19.0.0-beta-4508873393-20240430",
+    "waku": "workspace:*"
+  },
+  "devDependencies": {
+    "@types/react": "18.2.74",
+    "@types/react-dom": "18.2.24",
```
Comment (dai-shi) [inline e2e/fixtures/rsc-css-modules/package.json:19]:
```suggestion
    "@types/react": "18.3.1",
    "@types/react-dom": "18.3.0",
```
but maybe it's not related?

**Comment by dai-shi** [comment]
Ah, you need to add a reference in `tsconfig.e2e.json`.
If it works, feel free to improve CONTRIBUTING.md, thanks!

File: e2e/fixtures/rsc-css-modules/package.json (hunk 4 above)
Comment (negreirosleo) [reply e2e/fixtures/rsc-css-modules/package.json:19]: I updated it but chose to not push the commit until the test was 100% passing locally so I do not waste your CI

**Comment by negreirosleo** [comment]
I'm checking the final error so we can get this done

**Review by himself65 — APPROVED** [review]
LGTM, maybe we need to abstract e2e setup script so we don't need to copy paste everything

**Comment by negreirosleo** [comment]
> LGTM, maybe we need to abstract e2e setup script so we don't need to copy paste everything

I can work on that if you want

**Pull request merged**